src/librustc_mir/interpret/const_eval.rs

```diff
@@ -123,20 +123,23 @@ pub fn value_to_const_value<'a, 'tcx>(
             }
         }
         _ => {
-            if let Value::ByRef(ptr, align) = val {
+            if let Value::ByRef(ptr, _) = val {
                 let ptr = ptr.primval.to_ptr().unwrap();
                 assert_eq!(ptr.offset, 0);
-                let alloc = tcx.interpret_interner
-                               .get_alloc(ptr.alloc_id)
-                               .expect("miri allocation never successfully created");
-                assert_eq!(align, alloc.align);
+                let alloc = ecx.memory.get(ptr.alloc_id)?;
```

Comment:
doesn't this change simply turn a panic into an error? How does that fix the regression?

Reply:
The alloc ID we are trying to read is stored in the `EvalContext` and isn't in the global table.

Reply:
ah. makes sense. Is there any reason reading this should ever fail? If so, then this code is fine. Otherwise it should probably have the panic, right?

Reply:
This should really recurse into allocations like `mark_static_initialized` and could replace it by calling `value_to_const_value` at the end of const_eval.

Reply:
> ah. makes sense. Is there any reason reading this should ever fail?

I don't think this can fail. `Value::ByRef` here must always point to a valid `AllocId` right?

Reply:
Might point to an integer address in general, but not as the result of const_eval

Reply:
`value_to_const_value` is really just converting between different representation of Rust values, so it shouldn't fail. I've changed it to use `unwrap` so it will panic if it happens to fail.